src/pybind/mgr/cephadm/module.py

```diff
@@ -2507,7 +2507,20 @@ def generate_grafana_ds_config(hosts: List[str]) -> str:
         for dd in self.cache.get_daemons_by_service('prometheus'):
             prom_services.append(dd.hostname)
             deps.append(dd.name())
-        cert, pkey = create_self_signed_cert('Ceph', 'cephadm')
+
+        cert = self.get_store('grafana_crt')
+        pkey = self.get_store('grafana_key')
+        if cert and pkey:
+            try:
+                verify_tls(cert, pkey)
+            except ServerConfigException as e:
+                logger.warning('Provided grafana TLS certificates invalid: %s', str(e))
```

Comment:
Future work: We could perhaps add a check from serve() that looks whether there is a grafana_crt/key configured, and if so, verifies they are still valid.  If not, it can raise a health alert.  We probably want to be slightly clever so that we don't fetch the values from config-key on every iteration, though, and don't repeat the (expensive?) check if the crt hasn't changed.